urbit/urbit
# docket: Add docket agent, split into grid desk

Adds a docket agent that manages grid tiles. It handles:
- Serving the files && fetching globs (deprecating file-server & glob agents)
- Discovery of package metadata
- Removal of packages

To test:
- boot a galaxy and a star under that galaxy with the pill in this branch
- `|merge %base-dev our %base, =gem %init`
- `|mount %base-dev`
- `rm -rf zod/base-dev/*`
- `rsync -zr src/pkg/base-dev/* zod/base-dev`
- `|commit %base-dev`
- `|merge %grid our %base-dev`
- `|mount %grid`
- `rsync -zr src/pkg/grid/* zod/grid`
- `|commit %grid`
- `|public %grid`
On the star:
- `|install ~zod %grid`
Once install finishes, you should be able to load the star's url in your browser and see the grid interface

testing landscape:
- on the galaxy:
- `|merge %landscape our %base-dev`
- `|mount %landscape`
- `rsync -zr src/pkg/landscape/* zod/landscape`
- `|commit %landscape`
- `|install ~zod %landscape`
- load the star, type `~zod/landscape` in the omnibox and hit enter, it should give you a rendering of the packages metadata, from which you can click 'Get App' and it will install the app.


Notes:
- Does not currently work with devserver. I have the devserver working but `ted/dist` needs to merge master before I add those patches
- Should the `sur/` files for %hood be moved to base-dev?

Demo: 

https://user-images.githubusercontent.com/46801558/128978439-efed4285-3752-4ff3-a25d-4c864aef88d1.mov
